packages/hash/api/src/graphql/typeDefs/knowledge/entity.typedef.ts

```diff
@@ -1,116 +1,41 @@
-    """
-    ownedById: ID!
-    """
-    Alias of ownedById - the id of the account that owns this entity.
-    """
-    accountId: ID!
-      @deprecated(reason: "accountId is deprecated. Use ownedById instead.")
-    """
-    The versioned URI of this entity's type.
-    """
-    entityTypeId: String!
-    """
-    The JSON object containing the entity's properties.
-    """
-    properties: JSONObject!
-  }
+export const entityWithMetadataTypedef = gql`
+  scalar EntityId
+  scalar EntityEditionId
+  scalar PropertyObject
+  scalar EntityMetadata
 
-  type UnknownPersistedEntity implements PersistedEntity {
-    # ENTITY INTERFACE FIELDS BEGIN #
-    """
-    The id of the entity
+  type EntityWithMetadata {
     """
-    entityId: ID!
+    Metadata for the entity.
     """
-    The specific version of the entity
+    metadata: EntityMetadata!
     """
-    entityVersion: String!
+    Properties of entity.
     """
-    The id of the account that owns this entity.
-    """
-    ownedById: ID!
-    """
-    Alias of ownedById - the id of the account that owns this entity.
-    """
-    accountId: ID!
-      @deprecated(reason: "accountId is deprecated. Use ownedById instead.")
-    """
-    The versioned URI of this entity's type.
-    """
-    entityTypeId: String!
-    """
-    The JSON object containing the entity's properties.
-    """
-    properties: JSONObject!
-    # ENTITY INTERFACE FIELDS END #
-  }
-
-  """
-  For referring to an existing entity owned by a specific account id
-  """
-  input PersistedExistingEntity {
-    """
-    This may be a reference to a placeholder set using placeholderId on a previous UpdatePageContentsAction.
-    """
-    entityId: ID!
-    ownedById: ID!
-  }
-
-  """
-  Select entity types by ONE of componentId, entityTypeId
-  """
-  input PersistedEntityTypeChoice {
-    # Previously the EntityTypeChoice included 'componentId: ID', which made it possible
-    # to create a block using an already-existing entity type based on its componentId
-    # we should reconsider what we do about the component ID
-    # see https://app.asana.com/0/0/1202924026802716/f
-    """
-    A fixed entity type ID. This may be a reference to a placeholder set using a previous createEntityTypeAction.
-    """
-    entityTypeId: String
+    properties: PropertyObject!
   }
 
   input PersistedLinkedEntityDefinition {
     destinationAccountId: ID!
-    linkTypeId: String!
+    linkEntityTypeId: String!
     """
     The index of the link (if any)
     """
     index: Int
-    entity: PersistedEntityDefinition!
+    entity: EntityWithMetadataDefinition!
   }
 
-  input PersistedEntityDefinition {
+  input EntityWithMetadataDefinition {
     """
     Existing Entity to use instead of creating a new entity.
     """
-    existingEntity: PersistedExistingEntity
+    existingEntity: EntityId
     """
     The type of which to instantiate the new entity.
     """
-    entityType: EntityTypeWithMetadata
+    entityType: String
```

Comment:
Shouldn't this be `EntityType`?

Reply:
Can you update as necessary then :)

Reply:
String would be correct here. We don't have a `VerisonedUri` type/scalar. yet Do we want to introduce it here? I can export it from the type system package from the subgraph lib to get the template literal string

Reply:
I think all we want to do is rename the variable if that's the case. (Just `EntityTypeId` or something)

We should probably consider adding `VersionedUri` to the GraphQL types soon though, as otherwise we are going to have to mess around with casts or checks on either side of the boundary which will be a pain. I leave it up to you to decide if it's big enough to constitute its own task :)

Reply:
Introduced `VersionedUri` 9607d6d047f77b25db9016fa492c52d902bf98f7 and adjusted references

Reply:
I think codegen is broken on that commit?

Reply:
> @hashintel/hash-frontend:codegen: [FAILED]   Error 0: Variable "$dataTypeId" of type "String!" used in position expecting type "VersionedUri!".

Reply:
Fixed it locally on my branch while experimenting so I've pushed :) We'll hopefully get this turborepo caching problem fixed